eval.go

```diff
@@ -641,7 +641,8 @@ func (d *indexData) regexpToMatchTreeRecursive(r *syntax.Regexp, minTextSize int
 	case syntax.OpLiteral:
 		s := string(r.Rune)
 		if len(s) >= minTextSize {
-			mt, err := d.newSubstringMatchTree(&query.Substring{Pattern: s, FileName: fileName, CaseSensitive: caseSensitive})
+			ignoreCase := syntax.FoldCase == (r.Flags & syntax.FoldCase)
+			mt, err := d.newSubstringMatchTree(&query.Substring{Pattern: s, FileName: fileName, CaseSensitive: !ignoreCase && caseSensitive})
```

Comment:
this makes sense to me. When we actually run the regex against a document we set the FoldCase flag on the regex (by prefixing `(?i)` I believe). But if you have case sensitivity set and you put case insensitive in the regex syntax, then it is in effect case insensitive. So in all places we convert regex to literals we need to check this. Nice catch! I wonder if there are other parts where we need to do this? Off-hand I can't think of another code path, but worth investigating all places we read the regexp from a query struct.

Reply:
I dug in based on this feedback to see where else we ignore (?i) and I found a [place](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/zoekt@6df055493b1600051e467b79d296fa292f658d3f/-/blob/query/parse.go?L261) but not sure how to fix. 

During parsing here 
![image](https://github.com/sourcegraph/zoekt/assets/304410/4bf14232-970f-45a8-b402-a81f0b41dcc1)

We will parse the regex out and for some reason map it as a Substring and not as a regex and this loses the flags. Substring does have a caseSensitive property on it, but already today its defaulted to false and downstream code ignores it and lets the `case:yes` flag trump it. It seems we would need to plumb up a way to make this explicit or represent this as a regex so later code can check flags.

Either way, I am not sure its worth the change. Will close the PR for now but can re-open if I am over thinking it @keegancsmith  :)

Reply:
I think that is a place you should set CaseSensitive if ?i is not set in the regex? But yeah it is a bit tricky, since the way case: is implemented in zoekts query parser it will mutate CaseSensitive. I too would have to think a bunch more to know what actually makes sense to do here.

However, your original fix seems like it only makes us more correct even though we don't handle this everywhere.

Reply:
ok Ill send PR just for that one case then

Reply:
Added unit test now too @keegancsmith